app/routers/users.py

```diff
@@ -259,11 +262,66 @@ async def resend_user_invitation(
 
 
 @router.get("/{id}", response_model=UserRead)
-async def get_user_by_id(id: str, token: str | None = None):  # pragma: no cover
-    # if token is provided no authentication is needed but
-    # an AccountOperator in status invited must exist with
-    # user id and token and the token must not be expired
-    pass  # not yet implemented
+async def get_user_by_id(
+    id: str,
+    auth_context: CurrentAuthContext,
+    accountuser_repo: AccountUserRepository,
+    user_repo: UserRepository,
+    token: str | None = None,
+):
+    """
+        This endpoint returns the user filtered by the given ID.
+    There are 3 possible scenarios
+    1. No Authentication is provided, but an invitation token is sent with the request.
+       In this case, if the invitation token is valid and the user exists, its record will
+       be returned.
+    2. Authentication is provided, and the Account is OPERATIONS. In this case, the query will
+    be run with no more checks
+    3. Authentication is provided, and the Account is AFFILIATE. In this case, the query is run only
+    if the Account User status is not DELETED.
+
+    Raises:
+        - HTTPException with status 404 if no account user is found
+        - HTTPException 401 if the invitation token is not valid
+
+
+    """
+    user_id = id
+    response = None
+    if auth_context is None:
+        # No Authentication. We must verify the invitation token
+        invitation_token = token
+        account_user = await accountuser_repo.first(
+            AccountUser.invitation_token == invitation_token,
+            AccountUser.status.in_(
+                [AccountUserStatus.INVITED, AccountUserStatus.INVITATION_EXPIRED]
+            ),
+        )
+        if account_user is None:
+            raise UNAUTHORIZED_EXCEPTION
+        response = await user_repo.get(id=user_id)
+    elif auth_context.account.type == AccountType.OPERATIONS:
+        with wrap_exc_in_http_response(NotFoundError, "User Not Found."):
```

Comment:
In this case NotFoundError bring a standard message used then by `wrap_exc_in_http_response` (see [https://github.com/softwareone-platform/ffc-finops-operations/blob/main/app/utils.py#L22C5-L22C30](https://github.com/softwareone-platform/ffc-finops-operations/blob/main/app/utils.py#L22). What you have to pass to the `wrap_exc_in_http_response` is the status code you want to use since by default it use a 400. See [https://github.com/softwareone-platform/ffc-finops-operations/blob/main/app/routers/systems.py#L37](https://github.com/softwareone-platform/ffc-finops-operations/blob/main/app/routers/systems.py#L37)